src/root-logger.ts

```diff
@@ -1,295 +1,38 @@
-         * helpful for newcomers or a matter of taste for some.
-         */
-        levelLabel?: boolean
-        /**
-         * Should the logs include the time between it and previous log?
-         *
-         * @default `true`
-         */
-        timeDiff?: boolean
-      }
-  /**
-   * Toggle pieces of data that should or should not be logged.
-   */
-  data?: {
-    /**
-     * The Unix timestamp in milliseconds when the log was written to the
-     * output.
-     *
-     * @defualt `true` if NODE_ENV="production"
-     */
-    time?: boolean
-    /**
-     * The current node process ID assigned by the operating system. Acquired
-     * via `process.pid`.
-     *
-     * @defualt `true` if NODE_ENV="production"
-     */
-    pid?: boolean
-    /**
-     * The host name of the machine this process is running on. Acquired via `OS.hostname()`.
-     *
-     * @defualt `true` if NODE_ENV="production"
-     */
-    hostname?: boolean
-  }
-}
-
-type Settings = SettingsData & {
-  (newSettings: SettingsInput): RootLogger
+export type SettingsManager = Settings.Data & {
+  (newSettings: Settings.Input): RootLogger
 }
 
 // TODO jsDoc for each option
-export type Options = SettingsInput & {
-  output?: Output.Output
+export type Options = Settings.Input & {
   name?: string
 }
 
 export type RootLogger = Logger.Logger & {
-  settings: Settings
+  settings: SettingsManager
 }
 
 export type State = {
-  settings: Settings
+  settings: SettingsManager
 }
 
 /**
  * Create a root logger.
  */
 export function create(opts?: Options): RootLogger {
-  let level = opts?.level
-  if (level === undefined) {
-    if (process.env.LOG_LEVEL) {
-      level = parseFromEnvironment<Level.Level>('LOG_LEVEL', Level.parser)
-    } else {
-      level = process.env.NODE_ENV === 'production' ? Level.LEVELS.info.label : Level.LEVELS.debug.label
-    }
-  }
-
-  const settings = ((newSettings: SettingsInput) => {
-    if ('pretty' in newSettings) {
-      // @ts-ignore
-      logger.settings.pretty = processSettingInputPretty(newSettings.pretty, logger.settings.pretty)
-      // Sync chalk
-      // Assume true color support, not doing all that -> https://github.com/chalk/chalk#256-and-truecolor-color-support
-      chalk.level = logger.settings.pretty.color ? 3 : 0
-    }
-
-    if ('data' in newSettings) {
-      // @ts-ignore
-      logger.settings.data = processSettingInputData(newSettings.data, logger.settings.data)
-    }
-
-    if ('level' in newSettings) {
-      // @ts-ignore
-      logger.settings.level = newSettings.level
-    }
-
+  const settings = Settings.create(opts)
+  const settingsManager = ((newSettings) => {
+    settings(newSettings)
+    Object.assign(settingsManager, settings)
     return logger
-  }) as Settings
-
-  const state = { settings } as State
-  const loggerLink = Logger.create(state, [opts?.name ?? 'root'], {})
+  }) as SettingsManager
+  Object.assign(settingsManager, settings)
```

Comment:
Very unhappy with how this system/code looks. Open to refactoring further later.